### DISCUSSION TITLE
Media Queries for different Grid templateAreas?

### DISCUSSION THREAD

**Discussion by ghost** [chakra-ui/chakra-ui#2781] [Q&A] [answered]
In regular CSS, I can have different Grid Template Areas for different screen sizes.

For example:

`		grid-template-columns: 1fr;
		grid-template-rows: 2fr 2fr 1fr 1.5fr 0.5fr;
		grid-template-areas:
			'logo'
			'input'
			'submit'
			'remember'
			'bottom';

		text-align: center;
		justify-content: center;
		align-items: center;

		/* For wider screens */
		@media only screen and (orientation: landscape) {
			grid-template-columns: 4em 3fr 2fr 4em;
			grid-template-rows: 1fr 3fr 2fr 0.5fr;

			grid-template-areas:
				'logo 			logo				logo				logo'
				'.. 				input			submit			..'
				'..				remember		remember		..'
				'bottom			bottom			bottom			bottom';
	`

How can I achieve the same result in Chakra?

When I put the @media code inside a <Grid> it says the code is unreachable.

    <Grid
      h="100vh"
      bg="yellow.100"
      templateColumns="1fr"
      templateRows="2fr 2fr 1fr 1.5fr 0.5fr"
      templateAreas="   'logo'
                        'input'
                        'submit'
                        'remember'
                        'bottom'"
                        
		/* For wider screens */
		@media only screen and (orientation: landscape) {
			grid-template-columns: 4em 3fr 2fr 4em;
			grid-template-rows: 1fr 3fr 2fr 0.5fr;

			grid-template-areas:
				'logo 			logo				logo				logo'
				'.. 				input				submit			..'
				'..					remember		remember		..'
				'bottom			bottom			bottom			bottom';
		}                        
    >`

**Comment by chr-ge** [comment, marked as answer]
Take a look at the responsive style [Array](https://chakra-ui.com/docs/features/responsive-styles#the-array-syntax) or [Object](https://chakra-ui.com/docs/features/responsive-styles#the-object-syntax) syntax.
Here is an example on how you can use it:
```jsx
<Grid
  h="100vh"
  bg="yellow.100"
  templateColumns={{ base: "1fr", lg: "4em 3fr 2fr 4em" }}
  templateRows={{ base: "2fr 2fr 1fr 1.5fr 0.5fr", lg: "1fr 3fr 2fr 0.5fr" }}
  ...
>
  ...
</Grid>
```
[Here is what it's doing under the hood](https://chakra-ui.com/docs/features/responsive-styles#under-the-hood).

**Reply by ghost** [reply]
Thanks so much!

The part I'm confused about, though, is how to use that technique for area templates.

How can I base it on Orientation instead of Width?

How do I put something like THIS in that array method?

			grid-template-areas:
				'logo 			logo			logo			logo'
				'.. 			input			submit			..'
				'..			remember		remember		..'
				'bottom			bottom			bottom			bottom';

**Reply by chr-ge** [reply]
Try this:
```jsx
<Grid
  templateAreas={{ 
    base: '"logo logo logo logo" ".. input submit .." ".. remember remember .." "bottom bottom bottom bottom"', 
    lg: ...
  }}
  ...
>
  ...
</Grid>
```

**Comment by Sagar-Gondage** [comment]
Hi There, I hope this will help.
I have made Grid Responsive to at least FIve screen sizes as per Chakra-Ui dimensions(base, sm, md, lg, xl)

export const ResposiveGrid = () => {
  const Text = useBreakpointValue({
    base: { text: "Base Layout" },
    sm: { text: "Small Layout" },
    md: { text: "Medium Layout" },
    lg: { text: "Large Layout" },
    xl: { text: "Xtra Large Layout" }
  });
  return (
    <Container maxW="1280px" border="1px solid black">
      <Box fontWeight="bold">{Text.text}</Box>
      <Grid
        templateAreas={{
          base: `"NAV" "WIDGET" "CONTENT"`,
          sm: `"NAV CONTENT WIDGET"`,
          md: `"WIDGET WIDGET WIDGET"
               "CONTENT NAV NAV"
               "CONTENT NAV NAV"`,
          lg: `"CONTENT CONTENT NAV"
               "CONTENT CONTENT WIDGET"
               "CONTENT CONTENT WIDGET"`,
          xl: `"NAV NAV NAV"
               "CONTENT CONTENT WIDGET"
               "CONTENT CONTENT WIDGET"`
        }}
        gridTemplateRows={{
          base: "6rem 8rem 8rem",
          sm: "22rem",
          md: "6rem 8rem 8rem",
          lg: "6rem 8rem 8rem",
          xl: "6rem 8rem 8rem"
        }}
        gridTemplateColumns={{
          base: "100%",
          sm: "100px 1fr 100px",
          md: "1fr 1fr 1fr",
          lg: "2fr 1fr 1fr",
          xl: "60% 20% 1fr"
        }}
        gap="1"
        color="black"
        fontWeight="bold"
        pb={5}
      >
        <GridItem pl="2" bg="orange.300" area={"NAV"}>
          <Box>NAV</Box>
        </GridItem>
        <GridItem pl="2" bg="pink.300" area={"CONTENT"}>
          <Box>CONTENT</Box>
        </GridItem>
        <GridItem pl="2" bg="green.300" area={"WIDGET"}>
          <Box>WIDGET</Box>
        </GridItem>
      </Grid>
    </Container>
  );
};
